src/misc.c

```diff
@@ -159,23 +159,28 @@ _libssh2_recv(libssh2_socket_t sock, void *buffer, size_t length,
     (void)abstract;
 
     rc = recv(sock, buffer, length, flags);
+    if(rc < 0) {
 #ifdef WIN32
-    if(rc < 0)
-        return -wsa2errno();
+        int err = _libssh2_wsa2errno();
 #else
-    if(rc < 0) {
+        int err = errno;
+#endif
```

Comment:
I suggest deduping the `int err` declration into the top of the block. Also in the other similar cased in the patch:
```
... {
  int err;
#ifdef WIN32
  err = ...
#else
  err = ...
#endif
```

Reply:
Updated